torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -638,6 +638,11 @@ class TORCH_API ProcessGroupNCCL : public Backend {
 
   void performNocolorSplit(at::Device device);
 
+  c10::StreamId getStreamId(
+      OpType opType,
```

Comment:
I'm trying to understand the arguments in this new API, wondering why do we need `opType, peer, device`? IMO given that pg already associate with/hold a stream internally, this API should simply just return that stream?

Reply:
> pg already associate with/hold a stream internally

Not sure if I missed something but I believe it doesn't work that way today?
IIUC, the nccl communicator is lazily created, bound to the pairs (either collective or p2p) and cached (reused for future comms using the same pair as the current).

So to your question, the API reflects this property (see the cpp counterpart).